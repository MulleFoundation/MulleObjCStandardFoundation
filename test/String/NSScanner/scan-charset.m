//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCStandardFoundation/MulleObjCStandardFoundation.h>
//#import "MulleStandaloneObjCFoundation.h"

#include <stdio.h>


void  test( BOOL flag)
{
   NSString    *strings[] =
   {
      @"1848",
      @" 18 48 ",
      @"11a2bb",
      @""
   };
   NSCharacterSet    *sets[] =
   {
      [NSCharacterSet decimalDigitCharacterSet],
      [NSCharacterSet letterCharacterSet]
   };
   char    *setNames[] =
   {
      "decimalDigitCharacterSet",
      "letterCharacterSet"
   };
   NSCharacterSet    *set;
   NSScanner         *scanner;
   NSString          *string;
   NSString          *s1;
   NSString          *s2;
   int               i;
   int               j;
   char              *s;
   char              *sep;

   for( i = 0; i < sizeof( sets) / sizeof( sets[ 0]); i++)
   {
      set = sets[ i];
      printf( "%s (skip white: %s)\n", setNames[ i], flag ? "NO" : "YES");

      for( j = 0; j < sizeof( strings) / sizeof( strings[ 0]); j++)
      {
         string = strings[ j];
         sep    = "";
         printf( "\tScanning \"%s\" :: ", [string UTF8String]);
         scanner = [NSScanner scannerWithString:string];
         if( flag)
            [scanner setCharactersToBeSkipped:nil];

         while( ! [scanner isAtEnd])
         {
            [scanner scanCharactersFromSet:set
                                intoString:&s1];
            [scanner scanUpToCharactersFromSet:set
                                     intoString:&s2];
            printf( "%sscanSet:\"%s\" scanNotInSet:\"%s\"", sep, [s1 UTF8String], [s2 UTF8String]);
            sep =", ";
         }
         printf( "\n");
      }
   }
}


int main( int argc, const char * argv[])
{
   test( YES);
   test( NO);
   return( 0);
}

