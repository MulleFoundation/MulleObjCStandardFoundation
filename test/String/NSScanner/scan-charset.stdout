decimalDigitCharacterSet (skip white: NO)
	Scanning "1848" :: scanSet:"1848" scanNotInSet:""
	Scanning " 18 48 " :: scanSet:"" scanNotInSet:" ", scanSet:"18" scanNotInSet:" ", scanSet:"48" scanNotInSet:" "
	Scanning "11a2bb" :: scanSet:"11" scanNotInSet:"a", scanSet:"2" scanNotInSet:"bb"
	Scanning "" :: 
letterCharacterSet (skip white: NO)
	Scanning "1848" :: scanSet:"" scanNotInSet:"1848"
	Scanning " 18 48 " :: scanSet:"" scanNotInSet:" 18 48 "
	Scanning "11a2bb" :: scanSet:"" scanNotInSet:"11", scanSet:"a" scanNotInSet:"2", scanSet:"bb" scanNotInSet:""
	Scanning "" :: 
decimalDigitCharacterSet (skip white: YES)
	Scanning "1848" :: scanSet:"1848" scanNotInSet:""
	Scanning " 18 48 " :: scanSet:"18" scanNotInSet:"", scanSet:"48" scanNotInSet:""
	Scanning "11a2bb" :: scanSet:"11" scanNotInSet:"a", scanSet:"2" scanNotInSet:"bb"
	Scanning "" :: 
letterCharacterSet (skip white: YES)
	Scanning "1848" :: scanSet:"" scanNotInSet:"1848"
	Scanning " 18 48 " :: scanSet:"" scanNotInSet:"18 48 "
	Scanning "11a2bb" :: scanSet:"" scanNotInSet:"11", scanSet:"a" scanNotInSet:"2", scanSet:"bb" scanNotInSet:""
	Scanning "" :: 
